apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -103,6 +103,48 @@ Services.obs.addObserver(function(document) {
     });
   }
 
+  function getMessageNumbers(message) {
+    var numbers = [];
+
+    if (message.sender) {
+      numbers.push(message.sender);
+    }
+
+    if (message.receiver) {
+      numbers.push(message.receiver);
+    }
+
+    if (message.receivers) {
+      for(var i = 0; i < message.receivers.length; i++) {
+        numbers.push(message.receivers[i]);
+      }
+    }
+
+    return numbers;
+  }
+
+  function getMessagesFromFilter(threads, filter) {
+    var messages = [];
+
+    if (filter.threadId) {
+      messages = threads.get(filter.threadId).messages;
+    } else if (filter.numbers && filter.numbers.length) {
+      for (var thread of threads.values()) {
+        for(var message of thread.messages) {
```

Comment:
if "messages" is an array (I don't remember if it's an array or a map or a set :) ) you could use `.filter` (unless you decide to use mozPhoneNumberService which is asynchronous)